server/applicationset/applicationset.go

```diff
@@ -363,7 +363,14 @@ func (s *Server) Generate(ctx context.Context, q *applicationset.ApplicationSetG
 	if appset == nil {
 		return nil, errors.New("error creating ApplicationSets: ApplicationSets is nil in request")
 	}
-	namespace := s.appsetNamespaceOrDefault(appset.Namespace)
+	// The namespace explicitly to be set to the server namespace
+	// since this is the exact namespace that is being used
+	// to generate parameters (especially for git generator).
+	// In case of Git generator, if the namespace is set to
+	// appset namespace, we'll look for a project in the appset
+	// namespace that would lead to error when generating params
+	// for an appset in any namespace feature.
+	namespace := s.appsetNamespaceOrDefault(s.ns)
```

Comment:
```suggestion
	namespace := s.appsetNamespaceOrDefault(s.ns)
```
The reason behind changing the namespace to server namespace is because the same namespace is getting[ passed to generate parameters in case of git generator](https://github.com/argoproj/argo-cd/blob/02de363d9cc3cbdf4de0b56ff21eb6d4c7cc3ca1/applicationset/generators/git.go#L81) that's causing issue. When an appset namespace (such as `dev`) is passed, we try to look for a project in the appset namespace. In Argo CD, there's no concept of AppProject in any namespace hence it's better to start searching for the AppProject in the server namespace than appset namespace.

Reply:
That's a very good suggestion Michael! Can you review this now?